### ISSUE TITLE
Hit pages and threads

### ISSUE THREAD

**Issue by andrew2net** [riboseinc/gbbib#10]
@ronaldtse in IsoBib gem we have HitPages class which allow retrieving all search result pages. Actually, most relevant documents placed at the beginning of the first page. Asciidoctor-iso gem uses the only first page to search matched document.
In GbBib gem I implemented only HitCollection class which contains the first page of search results.
Also, I didn't implement threads and scraping all found pages at a time, since we don't utilize it in Asciidoctor-iso.
Do you think we need HitPages class, scrape all pages at a time and threads in GbBib gem?

**Comment by ronaldtse** [comment]
Agree, returning a HitCollection with first page results is good enough. Thanks!
